### PR TITLE
feat(lib): add schematics

### LINKED ISSUE #707: feat: add schematics

### PR DESCRIPTION
**Pull request by CaerusKaru**
Fixes #707

### CODE REVIEW

File: src/lib/schematics/install/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  let runner: SchematicTestRunner;
+  let appTree: Tree;
+
+  beforeEach(() => {
+    appTree = createTestApp();
+    runner = new SchematicTestRunner('schematics', collectionPath);
+  });
+
+  it('should update package.json', () => {
+    const tree = runner.runSchematic('ng-add', {}, appTree);
```
Comment (ThomasBurleson) [inline src/lib/schematics/install/index.spec.ts]: This is not testing for multiple apps nor for using the `--project` option.

File: src/lib/schematics/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Angular Flex Layout Schematics
+A collection of Schematics for Angular Flex Layout.
+
+## Collection
+
+### Install
+Adds Angular Flex Layout and its dependencies and pre-configures the application. 
+
+- Adds Flex Layout and CDK to `package.json`
+- Ensure `FlexLayoutModule` is installed and included in root module
```
Comment (ThomasBurleson) [inline src/lib/schematics/README.md]: Currently adding Flex-Layout is only supported for a specified application:

```console
ng add @angular/flex-layout --project=stellar-planner
```

We should add another option to add FlexLayout to a specified **module** using a `--module` option.

e.g.
```console
ng add @angular/flex-layout --module=libs/planner/feature-dashboard/src/lib/planner-dashboard.module.ts
```

File: src/lib/schematics/install/schema.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+export interface Schema {
+  /** Whether to skip package.json install. */
+  skipPackageJson: boolean;
+
+  /** Name of the project to target. */
+  project?: string;
+}
```
Comment (ThomasBurleson) [inline src/lib/schematics/install/schema.ts]: We need a `module?:string` option also.

**Comment by CaerusKaru** [comment]
I know this would be great for the community, but I don't think this is something we can support now or in the near-future. I know the tooling team wants to clean up the schematics libraries. If/when they do we can reconsider this. For now, I'm going to close this PR.

**Pull request closed without merging**